### ISSUE TITLE
question: google-authenticator PAM library provide option to disable/enable totp authentication runtime at user level

### ISSUE THREAD

**Issue by raviteja-b** [google/google-authenticator-libpam#257]
I have integrated google-authenticator-libpam into PAM configuration of system which has multiple users 
while one user wanted to have TOTP based google-authentication while other user wanted to have TOTP based google-authentication disabled.

does google-authenticator PAM library provide any option to enable/disable totp authentication runtime at user level ?

**Comment by ThomasHabets** [comment]
The easiest option is to simply use the `nullok` option.

If you want to enforce it, then you should probably use some other PAM module that has a list of users, and skip the GA PAM module based on that.

**Comment by raviteja-b** [comment]
> nullok

Yes, we are using nullok, does google-authenticator  pam module return any specific return code to identify if user authenticated without totp or secret key setup?

**Comment by ThomasHabets** [comment]
No, and in my opinion that belongs in a separate PAM module, since one could want this for many PAM modules.

Looks like there is one already: https://linux.die.net/man/8/pam_listfile

**Comment by raviteja-b** [comment]
> No, and in my opinion that belongs in a separate PAM module, since one could want this for many PAM modules.
> 
> Looks like there is one already: https://linux.die.net/man/8/pam_listfile

ok, as per google-authentication pam module what is the right way to disable totp based authentication for an user?

**Comment by ThomasHabets** [comment]
[Here you go](https://chatgpt.com/share/6703fba9-e434-800e-8136-95e65be36a44). Chatjippity didn't get it right on the first attempt, but the second one looks right.

Invert the `sense` or the `success=X default=Y` if you want to force it for everyone except the ones in the list.

**Comment by abhilashraju** [comment]
@ThomasHabets I can use this for skipping bypassed MFA users. 
Is it possible if GoogleAuthenticator can return something similar to password change requried code (PAM_NEW_AUTHTOK_REQD) for the secrete key. This will help us to identify first time login for a user and take him though generate secret key process.

**Comment by ThomasHabets** [comment]
Maybe. But I think someone like you, who has this use case, may need to come up with how to best do this.

It sounds to me like if we just want to return another code in the code path of a list of users, then it should be a different PAM module. Something like:

```
if in the list, skip 2
GA PAM check
skip 1
unconditionally return change password required
```

Or "in the list" could instead be "user's `$HOME/.google_authenticator` doesn't exist"
